lib/semantic_puppet/dependency/graph_node.rb

```diff
@@ -33,10 +33,10 @@ def populate_children(nodes)
       end
 
       # @api internal
-      # @return [{ String => SortedSet<GraphNode> }] the satisfactory
+      # @return [{ String => Array<GraphNode> }] the satisfactory
       #         dependency nodes
       def dependencies
-        @_dependencies ||= Hash.new { |h, k| h[k] = SortedSet.new }
+        @_dependencies ||= Hash.new { |h, k| h[k] = Array.new }
```

Comment:
Do we need to worry that `dependencies[module_name]` won't be sorted anymore?
First place that came to my mind was: https://github.com/puppetlabs/puppet/blob/1a13e0cf96c70b303492e684f9ccf4c38207b3dd/lib/puppet/module_tool/applications/installer.rb#L163

But we call `#max` there so it shouldn't matter in that specific case.

Reply:
That's a good point, seems like `dependencies` should return a sorted array. But we probably only want to sort if it if it was mutated since the last time it was sorted.

Reply:
https://github.com/puppetlabs/semantic_puppet/pull/38/files#diff-5b77e564dbad3d2b1cf050146955d368cfd532bdecdcfea9f67e600af8ca35d1R107 sorts the dependencies after adding them